### PR TITLE
templates and command line interface

### PR DESCRIPTION
**Pull request by isysd**
From the README:

## flaskswagger Command
This package now comes with a very simple command line interface: flaskswagger. This command can be used to build and update swagger specs for your flask apps from the command line or at build time.

```
flaskswagger -h
usage: flaskswagger [-h] [--template TEMPLATE] [--out-dir OUT_DIR] app

positional arguments:
  app                  the flask app to swaggerify

optional arguments:
  -h, --help           show this help message and exit
  --template TEMPLATE  template spec to start with
  --out-dir OUT_DIR    the directory to output to
```

For example, this can be used to build a swagger spec which can be served from your static directory. In the example below, we use the manually created swagger.json.manual as a template, and output to the `static/` directory.

```
flaskswagger server:app --template static/swagger.json.manual --out-dir static/
```

### CODE REVIEW

**Comment by atlithorn** [comment]
Nice!! Going to try this out over the weekend and get back to you

**Comment by atlithorn** [comment]
Could you have forgotten to push a file?

```
(env)nitro:flask-swagger atli$ flaskswagger -h
Traceback (most recent call last):
  File "/Users/atli/work/external/flask-swagger/env/bin/flaskswagger", line 9, in <module>
    load_entry_point('flask-swagger==0.2.11', 'console_scripts', 'flaskswagger')()
  File "/Users/atli/work/external/flask-swagger/env/lib/python2.7/site-packages/pkg_resources/__init__.py", line 552, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/Users/atli/work/external/flask-swagger/env/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2672, in load_entry_point
    return ep.load()
  File "/Users/atli/work/external/flask-swagger/env/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2345, in load
    return self.resolve()
  File "/Users/atli/work/external/flask-swagger/env/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2351, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
ImportError: No module named build_swagger_spec
```

File: flask_swagger.py [hunk 1]
```diff
@@ -114,6 +115,17 @@ def spec():
...
+    if template is not None:
+        for tkey, tval in template.items():
+            if tkey == 'definitions':
+                for k, v in tval.items():
+                    output['definitions'][k] = v
+            if tkey == 'paths':
+                for k, v in tval.items():
+                    output['paths'][k] = v
+            else:
+                output[tkey] = tval
```
Comment (atlithorn) [inline flask_swagger.py:128]: How about this?

```
output =  {
   "swagger": "2.0",
   ...
   # don't define paths or definitions in output
}
output.update(template)
paths = defaultdict(dict)
definitions = defaultdict(dict)
# check for template provided paths and definitions
for k,v in output.get('paths',{}).iteritems():
   paths[k] = v
for k,v in output.get('definitions',{}).iteritems():
   definitions[k] = v
#at any point before returning
output["paths"] = paths
output["definitions"] = definitions
```

**Comment by isysd** [comment]
I did completely forget a file, thank you. :+1: 

Your alternate template merging suggestion looks good. This is my first time working with defaultdict and I wasn't sure how best to cope.

**Comment by atlithorn** [comment]
@isysd how would you run this for the example app in /examples? I've been trying the following with no luck
```
flaskswagger example:app
Traceback (most recent call last):
  File "/Users/atli/work/flask-swagger/env/bin/flaskswagger", line 9, in <module>
    load_entry_point('flask-swagger==0.2.11', 'console_scripts', 'flaskswagger')()
  File "/Users/atli/work/flask-swagger/env/lib/python2.7/site-packages/pkg_resources/__init__.py", line 552, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/Users/atli/work/flask-swagger/env/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2672, in load_entry_point
    return ep.load()
  File "/Users/atli/work/flask-swagger/env/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2345, in load
    return self.resolve()
  File "/Users/atli/work/flask-swagger/env/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2351, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "build/bdist.macosx-10.10-x86_64/egg/build_swagger_spec.py", line 27, in <module>
  File "build/bdist.macosx-10.10-x86_64/egg/build_swagger_spec.py", line 14, in run
  File "/Users/atli/work/flask-swagger/env/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2345, in load
    return self.resolve()
  File "/Users/atli/work/flask-swagger/env/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2351, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
ImportError: No module named example
```

**Comment by isysd** [comment]
The script is using the Python namespace to resolve the 'app' argument. I would think that simply adding an `__init__.py` file to ./examples would do it, but it doesn't. I do know it works with installed packages. I.e. if you add ./examples to setup.py, then install, then run flaskswagger again, it should work.

**Pull request merged**